components/script/dom/xrframe.rs

```diff
@@ -55,16 +53,21 @@ impl XRFrameMethods for XRFrame {
 
     /// https://immersive-web.github.io/webxr/#dom-xrframe-getviewerpose
     fn GetViewerPose(&self, reference: &XRReferenceSpace) -> Option<DomRoot<XRViewerPose>> {
-        if let Some(_) = reference.downcast::<XRStationaryReferenceSpace>() {
-            // For 3DOF devices all three kinds of reference spaces are identical
-            // FIXME(#23070, Manishearth) support originOffset
-            let left = XRView::new(&self.global(), &self.session, XREye::Left, &self.data);
-            let right = XRView::new(&self.global(), &self.session, XREye::Right, &self.data);
-            Some(XRViewerPose::new(&self.global(), &left, &right))
-        } else {
-            // FIXME(#23070, Manishearth) support identity reference spaces
-            // depends on https://github.com/immersive-web/webxr/issues/565
-            None
-        }
+        let pose = reference.get_viewer_pose(&self.data);
+        let left = XRView::new(
+            &self.global(),
```

Comment:
This is building new `XRView` objects every time it is called? Is that required by the spec, or are we allowed to update in place?

Reply:
Chrome updates in-place. This change can be made.

Reply:
Actually, I'd rather not fix this just yet, I'll bring it up on the spec. This is observable.

Reply:
https://github.com/immersive-web/webxr/issues/572